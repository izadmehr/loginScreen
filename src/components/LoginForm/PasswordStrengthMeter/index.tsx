import React from "react";
import { ZXCVBNScore } from "zxcvbn";

import PasswordStrengthMeterContianer from "./Styles";

interface Props {
  score: ZXCVBNScore;
  showMeterLabel: boolean;
  notSecure: boolean;
}
function createPasswordLabel(score: ZXCVBNScore): string {
  switch (score) {
    case 0:
      return "Weak";
    case 1:
      return "Weak";
    case 2:
      return "Fair";
    case 3:
      return "Good";
    case 4:
      return "Strong";
    default:
      return "Weak";
  }
}

function PasswordStrengthMeter(props: Props): JSX.Element {
  const { score, showMeterLabel } = props;
  const passwordLabel = createPasswordLabel(score);

  return (
    <PasswordStrengthMeterContianer>
      {showMeterLabel && (
        <progress
          className={`password-strength-meter-progress strength-${passwordLabel}`}
          value={score}
          max="4"
        />
      )}
      <br />
      <div className="password-strength-meter-label">
        {showMeterLabel && (
          <>
            <strong>Password strength:</strong> {passwordLabel}
          </>
        )}
      </div>
    </PasswordStrengthMeterContianer>
  );
}

export default PasswordStrengthMeter;
