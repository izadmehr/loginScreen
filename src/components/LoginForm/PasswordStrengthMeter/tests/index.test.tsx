import React from "react";
import renderer from "react-test-renderer";

import PasswordStrengthMeter, { createPasswordLabel } from "../index";

test("PasswordStrengthMeter render week password", () => {
  const tree = renderer
    .create(<PasswordStrengthMeter score={1} showMeterLabel notSecure />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("PasswordStrengthMeter render strong password", () => {
  const tree = renderer
    .create(
      <PasswordStrengthMeter score={4} showMeterLabel notSecure={false} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("createPasswordLabel result", async () => {
  expect(createPasswordLabel(0)).toBe("Weak");
  expect(createPasswordLabel(1)).toBe("Weak");
  expect(createPasswordLabel(2)).toBe("Fair");
  expect(createPasswordLabel(3)).toBe("Good");
  expect(createPasswordLabel(4)).toBe("Strong");
});
