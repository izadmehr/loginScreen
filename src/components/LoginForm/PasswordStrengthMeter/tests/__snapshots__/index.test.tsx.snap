// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordStrengthMeter render strong password 1`] = `
<div
  className="sc-bdVaJa eqAilR"
>
  <progress
    className="password-strength-meter-progress strength-Strong"
    max="4"
    value={4}
  />
  <br />
  <div
    className="password-strength-meter-label"
  >
    <strong>
      Password strength: 
      <p
        className="sc-bwzfXH dHbFZD"
      >
        Strong
      </p>
    </strong>
  </div>
</div>
`;

exports[`PasswordStrengthMeter render week password 1`] = `
<div
  className="sc-bdVaJa eqAilR"
>
  <progress
    className="password-strength-meter-progress strength-Weak"
    max="4"
    value={1}
  />
  <br />
  <div
    className="password-strength-meter-label"
  >
    <strong>
      Password strength: 
      <p
        className="sc-bwzfXH iVpbWn"
      >
        Weak
      </p>
    </strong>
  </div>
</div>
`;
